core/domain/learner_progress_services.py

```diff
@@ -447,41 +508,62 @@ def get_all_completed_collection_ids(user_id):
     account of new addition of explorations.
 
     Args:
         user_id: str. The id of the learner.
+        collection_summaries: list(CollectionSummary). The list of collection
+            summary domain objects to be filtered.
+        collection_ids: list(str). The ids of the collection corresponding to
+            the collection summary domain objects.
 
     Returns:
-        list(CollectionSummary). A list with the summary domain objects of the
-            completed collections.
-        int. The number of explorations deleted from the list as they are no
-            longer present.
-        int. The number of collections moved to the incomplete section on
+        tuple. A 3-tuple whose elements are as follows:
+        - list(CollectionSummary). A filtered list with the summary domain
+            objects of the completed collections.
+        - list(str). The ids of the collections that are no longer present.
+        - list(CollectionSummary). The summaries corresponding to those
+            collections which have been moved to the in progress section on
             account of new explorations being added to them.
     """
-    completed_collection_ids = get_all_completed_collection_ids(user_id)
-
-    number_deleted = 0
+    index = 0
+    nonexistent_completed_collection_ids = []
     completed_to_incomplete_collections = []
-    for collection_id in completed_collection_ids:
-        if not collection_services.does_collection_exists(collection_id):
-            number_deleted = number_deleted + 1
-            remove_collection_from_completed_list(user_id, collection_id)
-        elif collection_services.get_next_exploration_ids_to_complete_by_user(
-                user_id, collection_id):
-            remove_collection_from_completed_list(user_id, collection_id)
-            mark_collection_as_incomplete(user_id, collection_id)
-            completed_to_incomplete_collections.append(
-                collection_services.get_collection_titles_and_categories(
-                    [collection_id])[collection_id]['title'])
-
-    return collection_services.get_collection_summaries_matching_ids(
-        completed_collection_ids), number_deleted, completed_to_incomplete_collections # pylint: disable=line-too-long
+    filtered_completed_collection_summaries = []
+
+    completed_collections = collection_services.get_multiple_collections_by_id(
+        collection_ids, strict=False)
+
+    exploration_ids_completed_in_collections = (
+        collection_services.get_explorations_completed_in_collections(
+            user_id, collection_ids))
+
+    for collection_summary in collection_summaries:
+        if collection_summary is None:
+            nonexistent_completed_collection_ids.append(collection_ids[index])
+        else:
+            completed_exploration_ids = (
+                exploration_ids_completed_in_collections[index])
+            if len(
```

Comment:
No need len() here. Also pull this out into separate var (next_exploration_ids) -- that will address the line-too-long. In general try to avoid pylint:disable unless absolutely necessary.